src/Fetch.js

```diff
@@ -345,6 +345,15 @@ function fetchXHR(fetch, onsuccess, onerror, onprogress, onreadystatechange) {
     {{{ makeSetValue('fetch', C_STRUCTS.emscripten_fetch_t.readyState, 'xhr.readyState', 'i16') }}}
     {{{ makeSetValue('fetch', C_STRUCTS.emscripten_fetch_t.status, 'xhr.status', 'i16') }}}
     if (xhr.statusText) stringToUTF8(xhr.statusText, fetch + {{{ C_STRUCTS.emscripten_fetch_t.statusText }}}, 64);
+    if (fetchAttrSynchronous)
+      // The response url pointer malloc()ed here has the same lifetime as the emscripten_fetch_t structure itself has, and is
+      // freed when emscripten_fetch_close() is called.
+      var ruPtr = stringToNewUTF8(xhr.responseURL);
+      if (ruPtr) {
+        _free(fetch + {{{ C_STRUCTS.emscripten_fetch_t.responseUrl }}});
```

Comment:
I don't think you need to free here do you since the pointer will always be null when a request starts?

Reply:
Also I think you can skip this `if` check and just assign to responseUrl unconditionally (here and below).